idl/chromadb/proto/chroma.proto

```diff
@@ -46,6 +46,7 @@ message Segment {
     optional string collection = 5;
     optional UpdateMetadata metadata = 6;
     map<string,FilePaths> file_paths = 7;
+    optional string configuration_json_str = 8;
```

Comment:
lets just add a TODO for configuration to be strongly typed, there is a bit of a refactor involved but there are downsides to doing a json str in that we can't be informed of dropped fields etc inside of the config

Reply:
The strong version of this is to have separate Segment proto types, with their own config types within. That's a larger refactor; I will add an issue for it.

Reply:
Opened an issue: https://github.com/chroma-core/chroma/issues/2598